diff --git a/core/helper.js b/core/helper.js
index 9bf1070793927c516d03012f101786332ab1901c..62fca1af3a01ff0ee099cb67bb2d0d264caa803e 100644
--- a/core/helper.js
+++ b/core/helper.js
@@ -417,8 +417,8 @@ function markInvalidNestingInRoot(doc, root, rootTrackId, getContent) {
 	});
 	droppedElements.forEach(element => {
 		const id = element.getAttribute(NESTING_TRACK_ID_ATTRIBUTE_NAME);
-		element.prepend(doc.createComment(NESTING_START_MARKER + id));
-		element.append(doc.createComment(NESTING_END_MARKER + id));
+		Element.prototype.prepend.call(element, doc.createComment(NESTING_START_MARKER + id));
+		Element.prototype.append.call(element, doc.createComment(NESTING_END_MARKER + id));
 	});
 	setNestingMarkersData(root);
 	if (rootTrackId) {
@@ -457,7 +457,7 @@ function markInvalidNestingInRoot(doc, root, rootTrackId, getContent) {
 	}
 
 	function testDroppedElement(element) {
-		if (element.namespaceURI != HTML_NAMESPACE) {
+		if (element.namespaceURI != HTML_NAMESPACE || RAW_TEXT_TAG_NAMES.includes(element.tagName.toUpperCase())) {
 			return false;
 		}
 		let ancestor = element.parentElement;
@@ -628,7 +628,7 @@ function fixInvalidNesting(document, NESTING_TRACK_ID_ATTRIBUTE_NAME, preventCle
 					if (originalElement.contains(element)) {
 						element.replaceWith(...element.childNodes);
 					} else {
-						originalElement.append(...element.childNodes);
+						Element.prototype.append.call(originalElement, ...element.childNodes);
 						element.remove();
 					}
 				}
@@ -922,7 +922,7 @@ function getResourcesInfo(win, doc, element, options, data, elementHidden, compu
 			}
 		});
 	}
-	if (tagName == "SCRIPT") {
+	if (tagName == "SCRIPT" && !options.blockScripts) {
 		if (element.async && element.getAttribute("async") != "" && element.getAttribute("async") != "async") {
 			element.setAttribute(ASYNC_SCRIPT_ATTRIBUTE_NAME, "");
 			data.markedElements.push(element);
diff --git a/core/index.js b/core/index.js
index 490b3c11191cd7bed294526286b10a584ab9c82d..13338c038e50c41b8dd506777940601644579938 100644
--- a/core/index.js
+++ b/core/index.js
@@ -622,7 +622,7 @@ class Processor {
 		}
 		const collapsedElementCount = collapseRecreatedElements(this.doc);
 		const invalidNesting = collapsedElementCount || this.doc.querySelector(`[${util.NESTING_TRACK_ID_ATTRIBUTE_NAME}]`);
-		if (invalidNesting) {
+		if (invalidNesting && !this.options.blockScripts) {
 			const scriptElement = this.doc.createElement("script");
 			scriptElement.textContent = `(${util.getFixInvalidNestingSource()})(document, "${util.NESTING_TRACK_ID_ATTRIBUTE_NAME}");`;
 			this.doc.body.appendChild(scriptElement);
